resources/sts/4.12/hypershift/sts_hcp_installer_permission_policy.json

```diff
@@ -176,38 +178,126 @@
             }
         },
         {
-          "Sid": "RunInstancesRedHatOwnedAMIs",
-          "Effect": "Allow",
-          "Action": [
-            "ec2:RunInstances"
-          ],
-          "Resource": [
-            "arn:aws:ec2:*:*:image/*"
-          ],
-          "Condition": {
-            "StringEquals": {
-              "ec2:Owner": [
-                "531415883065",
-                "251351625822",
-                "210686502322"
-              ]
+            "Sid": "RunInstancesRedHatOwnedAMIs",
+            "Effect": "Allow",
+            "Action": [
+                "ec2:RunInstances"
+            ],
+            "Resource": [
+                "arn:aws:ec2:*:*:image/*"
+            ],
+            "Condition": {
+                "StringEquals": {
+                    "ec2:Owner": [
+                        "531415883065",
+                        "251351625822",
+                        "210686502322"
+                    ]
+                }
+            }
+        },
+        {
+            "Sid": "ManageInstancesRestrictedResourceTag",
+            "Effect": "Allow",
+            "Action": [
+                "ec2:TerminateInstances",
+                "ec2:GetConsoleOutput"
+            ],
+            "Resource": "arn:aws:ec2:*:*:instance/*",
+            "Condition": {
+                "StringEquals": {
+                    "aws:ResourceTag/red-hat-managed": "true"
+                }
             }
-          }
         },
         {
+            "Sid": "CreateGrantRestrictedResourceTag",
             "Effect": "Allow",
             "Action": [
-                "kms:CreateGrant"
+                "kms:CreateGrant",
+                "kms:GenerateDataKey*"
```

Comment:
Can you update this to `GenerateDataKeyWithoutPlaintext` 

See [here](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/EBSEncryption.html)